File: Python/Climbing Stairs.py
#!/usr/bin/env python

class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        dp = [0, 1]
        for i in range(2, 20):
            dp.append(dp[i-1] + dp[i-2])
        return dp[n]

File: C++/Climbing-Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 3) return n;
        vector<int> res(n);
        res[0] = 1, res[1] = 2;
        for (int i = 2; i < n; i ++) {
            res[i] = res[i - 1] + res[i - 2];
        }
        return res[n - 1];
    }
};